includes/payment-methods/class-wc-stripe-payment-request.php

```diff
@@ -1359,12 +1370,12 @@ public function ajax_create_order() {
 			define( 'WOOCOMMERCE_CHECKOUT', true );
 		}
 
-		// In case the state is required, but is missing, add a more descriptive error notice.
-		$this->validate_state();
-
 		// Normalizes billing and shipping state values.
 		$this->normalize_state();
 
+		// In case the state is required, but is missing, add a more descriptive error notice.
+		$this->validate_state();
+
```

Comment:
If we don't normalize before we validate the WooCommerce Checkout process will fail. I'm honestly not sure why.
`validate_state` doesn't interact with the `WC()->checkout()->process_checkout()` call AFAICT, but the normalization is ignored if we call `validate_state()` first. @ricardo do you have any ideas as to why?

Reply:
> do you have any ideas as to why?

I reproduced the issue here and it's not that the normalization is ignored, but `validate_state` is throwing an error before the normalization happens, because `billing_state` and/or `shipping_state` are empty.

`validate_state` is used to display a friendlier error message for countries where the Payment Request API doesn't offer a state field, but WooCommerce expects it (so the country is not supported). The core team is working on a better approach for a more loose validation for Payment Request orders p7bje6-3fv-p2.

It's not really intuitive, but I think it's fine to call `normalize_state` before `validate_state` for this fix. Or do the Hong Kong mapping outside and before both methods.

I can't wrap my mind around "region" being passed as postcode from Apple Pay. To me it should be called "Postal code". It's really confusing to users 😪

Reply:
On second thought. Apple Pay is not offering a state field, much like many other countries.

I think we should confirm if Hong Kong buyers understand the keyword "Region" as postal code, or if this is some sort of i18n issue from Apple Pay for Safari in English, because for most countries it says "Postal code" in the field on that position.

In that case, we should be validating that field as postal code indeed, instead of implementing this change.

Reply:
I'm just waiting for the Apple Pay devs to set up a call with us so we can discuss this in more detail. Thanks for taking a look! I'll come back to this once I've had a chance to talk to Apple Pay 👍